actionpack/test/controller/parameters/accessors_test.rb

```diff
@@ -419,4 +418,37 @@ class ParametersAccessorsTest < ActiveSupport::TestCase
     @params.dig(:person, :addresses)[0] = { city: "Boston", state: "Massachusetts" }
     assert_equal "Boston", @params.dig(:person, :addresses, 0, :city)
   end
+
+  test "has_value? no deprecation warning after the value is converted" do
+    params = ActionController::Parameters.new(foo: { bar: "baz" })
+    assert params.has_value?("bar" => "baz")
+    params[:foo] # converts value to AC::Params
+    assert_not_deprecated do
+      assert params.has_value?("bar" => "baz")
+    end
+  end
+
+  test "has_value? works with hash argument after conversion" do
+    without_deprecated_parameters_hash_equality do
+      params = ActionController::Parameters.new(foo: { bar: "baz" })
+      assert params.has_value?("bar" => "baz")
+      params[:foo] # converts value to AC::Params
+      assert params.has_value?("bar" => "baz")
+    end
+  end
+
+  test "has_value? works with AC:Parameters argument before conversion" do
+    without_deprecated_parameters_hash_equality do
+      params = ActionController::Parameters.new(foo: { bar: "baz" })
+      assert params.has_value?(ActionController::Parameters.new("bar" => "baz"))
+    end
+  end
+
+  private
+    def without_deprecated_parameters_hash_equality
```

Comment:
This name was confusing me because the default is true and the opt-in is false. Usually it is the other way around. Let's name this instead:
```suggestion
    def with_parameters_hash_equality_deprecation
```